### PR TITLE
feat: Add support for multi-column ALTER TABLE ADD COLUMN statements

### LINKED ISSUE #244: Support for Multi-Column ALTER TABLE ADD COLUMN
```sql
ALTER TABLE IF EXISTS table1
ADD COLUMN
  col1 VARCHAR(512),
  col2 VARCHAR(512),
  col3 VARCHAR(128);
```
Executing Queries like these results in `ParserError`. Please add support for these queries.

### PR DESCRIPTION
**Pull request by ca0904**
# feat: Add support for multi-column ALTER TABLE ADD COLUMN statements

## Overview

This PR introduces support for Snowflake's multi-column `ALTER TABLE ADD COLUMN` syntax, allowing users to add multiple columns in a single statement. The implementation automatically splits these statements into individual DuckDB-compatible `ALTER TABLE` commands.

Resolves #244

### 🏗️ New DDL Module Structure

- **Created `fakesnow/transforms/ddl.py`**: A dedicated module for DDL (Data Definition Language) transformations
- **Added `alter_table_add_multiple_columns` function**: Handles multi-column ALTER TABLE statements
- **Future-ready**: The new module provides a foundation for extending DDL support

### ✨ Multi-Column ALTER TABLE Support

- **Snowflake Syntax**: Now supports `ALTER TABLE tab1 ADD COLUMN col1 INT, col2 VARCHAR(50), col3 BOOLEAN;`
- **DuckDB Compatibility**: Automatically splits into separate statements:

  ```sql
  ALTER TABLE tab1 ADD COLUMN col1 INT;
  ALTER TABLE tab1 ADD COLUMN col2 VARCHAR(50);  
  ALTER TABLE tab1 ADD COLUMN col3 BOOLEAN;
  ```

- **Robust Parsing**: Handles complex column definitions including constraints and Snowflake-specific types
- **Clean Integration**: Works seamlessly with existing transform pipeline

### 🔧 Integration & Architecture

- **Transform Pipeline**: Integrated into the existing `_transform_explode` method in `cursor.py`
- **Import Structure**: Updated `transforms/__init__.py` to expose the function from the new DDL module
- **Backward Compatibility**: All existing imports and functionality remain unchanged

### ✅ Testing & Quality

- **Comprehensive Tests**: Added test coverage for multi-column ALTER TABLE scenarios
- **Edge Case Coverage**: Tests include IF EXISTS clauses, complex types, and schema-qualified tables
- **All Tests Pass**: Existing functionality remains unaffected

## Technical Details

The implementation handles Alter expressions with multiple ColumnDef actions:

- **Snowflake Dialect**: Uses SQLGlot's Snowflake dialect for parsing
- **Expression Splitting**: Splits multi-column ALTER statements into individual commands
- **IF EXISTS Support**: Handles optional IF EXISTS clauses
- **Type Preservation**: Maintains all column attributes and constraints

## Example Usage

**Before** (would fail):

```sql
ALTER TABLE users ADD COLUMN email VARCHAR(255), phone VARCHAR(20), created_at TIMESTAMP;
```

**After** (works seamlessly):

```sql
-- Input: Single multi-column statement
ALTER TABLE users ADD COLUMN email VARCHAR(255), phone VARCHAR(20), created_at TIMESTAMP;

-- Automatically transformed to:
ALTER TABLE users ADD COLUMN email VARCHAR(255);
ALTER TABLE users ADD COLUMN phone VARCHAR(20);
ALTER TABLE users ADD COLUMN created_at TIMESTAMP;
```

## Files Changed

- `fakesnow/transforms/ddl.py` - New DDL transformations module
- `fakesnow/transforms/__init__.py` - Updated imports
- `fakesnow/cursor.py` - Enhanced transform pipeline
- `tests/test_ddl.py` - Test coverage for new functionality

## Backward Compatibility

✅ **Fully backward compatible** - All existing functionality preserved
✅ **No breaking changes** - Existing imports and APIs unchanged
✅ **Progressive enhancement** - New syntax support without affecting existing queries

## Future Enhancements

The new `ddl.py` module provides a foundation for additional DDL transformations:

- CREATE TABLE enhancements
- DROP TABLE transformations
- ALTER TABLE modifications (beyond ADD COLUMN)
- INDEX operations
- VIEW operations
- CONSTRAINT management

---

This PR establishes a solid foundation for future DDL enhancements while providing immediate value through multi-column ALTER TABLE support.

### CODE REVIEW

**Comment by ca0904** [comment]
Currently `fakesnow` doesn't support this syntax right?
```sql
ALTER TABLE tab1 ADD COLUMN IF NOT EXISTS col1
```

**Comment by tekumara** [comment]
Yeh not yet but let's do that in a separate PR because it applies to the standard case without multiple columns too.

**Comment by ca0904** [comment]
Yeah sure! I was just curious about this. I have made no changes related to this.

File: fakesnow/transforms/ddl.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    DuckDB requires separate statements for each column.
+
+    Args:
+        expression: The expression to potentially transform
+
+    Returns:
+        List of expressions - multiple ALTER statements if transformation applied,
+        otherwise single original expression
+    """
+    if isinstance(expression, exp.Command):
```
Comment (tekumara) [inline fakesnow/transforms/ddl.py:33]: is this to handle `ALTER TABLE .. IF EXISTS` ? we can raise this as a sqlglot request, ideally we wouldn't have to do this parsing ourselves

File: tests/test_ddl.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import sqlglot
+
+from fakesnow.transforms.ddl import alter_table_add_multiple_columns
+
+
+def test_alter_table_add_multiple_columns() -> None:
+    # Test multiple columns are split into separate statements
+    sql = "alter table tab1 add column col1 int, col2 varchar(50), col3 boolean"
+    result = alter_table_add_multiple_columns(sqlglot.parse_one(sql))
```
Comment (tekumara) [inline tests/test_ddl.py:9]: this will make sure it's parsed as exp.Alter
```suggestion
    result = alter_table_add_multiple_columns(sqlglot.parse_one(sql, read="snowflake"))
```

File: tests/test_ddl.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    assert len(result) == 3
+    assert result[0].sql() == "ALTER TABLE tab1 ADD COLUMN col1 INT"
+    assert result[1].sql() == "ALTER TABLE tab1 ADD COLUMN col2 VARCHAR(50)"
+    assert result[2].sql() == "ALTER TABLE tab1 ADD COLUMN col3 BOOLEAN"
+
+    # Test multiple columns with IF EXISTS clause
+    sql_if_exists = "alter table if exists tab1 add column col1 int, col2 varchar(50)"
+    result_if_exists = alter_table_add_multiple_columns(sqlglot.parse_one(sql_if_exists))
```
Comment (tekumara) [inline tests/test_ddl.py:18]: same as above, let's parse it how it will be parsed inside cursor.py

**Review by tekumara — COMMENTED** [review]
looks great! could we add a small integration test to the existing `test_alter_table`?

File: fakesnow/transforms/ddl.py (hunk 1 above)
Comment (tekumara) [reply fakesnow/transforms/ddl.py:33]: we can leave this in for now tho, and remove it once sqlglot has support

Comment (ca0904) [reply fakesnow/transforms/ddl.py:33]: I added that as fail safe, if `sqlglot` fail to parse like this case.
But I didn't find any case fallback to `Command` so I am removing this.

**Comment by ca0904** [comment]
I am also thinking, we can make a PR template for future use. It will be easier to write PR message in future.

**Comment by tekumara** [comment]
I've added an integration test, thanks for the contribution @ca0904! 

> I am also thinking, we can make a PR template for future use. It will be easier to write PR message in future.

I've used PR templates in the past, but folks don't tend to follow them so they haven't worked so great.
